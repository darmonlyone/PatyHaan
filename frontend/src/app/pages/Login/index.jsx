import React from "react";
import { withRouter } from "react-router";

import PartyHaanImage from "app/asset/partyhaan.png";
import LinkButton from "app/common/LinkButton";
import InputField from "app/common/InputField";

import useLoginHandle from "app/hooks/useLoginHandle";

const Login = ({ history }) => {
	const {
		email,
		password,
		onEmailChange,
		onPasswordChange,
		submitLogin,
	} = useLoginHandle(history);

	return (
		<div className="flex justify-center items-center w-screen heightScreen">
			<div className="w-full px-10 flex flex-col sm:px-5 sm:w-96 sm:py-4">
				<img
					className="w-36 h-36 sm:w-44 sm:h-44 mx-auto mb-10"
					src={PartyHaanImage}
					alt="partyhaan logo"
				/>
				<InputField title="อีเมล" value={email} onChange={onEmailChange} />
				<InputField
					title="รหัสผ่าน"
					type="password"
					value={password}
					onChange={onPasswordChange}
				/>
				<div className="flex flex-col sm:flex-row mt-2 sm:mt-4">
					<LinkButton
						onClick={submitLogin}
						className="w-full my-2 mr-0 sm:my-0 sm:mr-2"
					>
						เข้าสู่ระบบ
					</LinkButton>
					<LinkButton
						to="/register"
						className="w-full my-2 ml-0 sm:my-0 sm:ml-2 bg-blue-400 hover:bg-blue-500"
					>
						สร้างบัญชีผู้ใช้
					</LinkButton>
				</div>
			</div>
		</div>
	);
};

export default withRouter(Login);
