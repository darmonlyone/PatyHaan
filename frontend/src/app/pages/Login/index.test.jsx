import React from "react";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";

import Login from ".";

it("should render Login page with default value", async () => {
	const { asFragment,  } = render(
		<Router>
			<Login />
		</Router>
	);

	expect(asFragment()).toMatchSnapshot();
});
