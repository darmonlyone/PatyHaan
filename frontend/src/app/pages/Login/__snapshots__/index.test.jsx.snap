// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Login page with default value 1`] = `
<DocumentFragment>
  <div
    class="flex justify-center items-center w-screen heightScreen"
  >
    <div
      class="w-full px-10 flex flex-col sm:px-5 sm:w-96 sm:py-4"
    >
      <img
        alt="partyhaan logo"
        class="w-36 h-36 sm:w-44 sm:h-44 mx-auto mb-10"
        src="partyhaan.png"
      />
      <div
        class="my-2 w-full"
      >
        <span
          class="mb-2 text-sm"
        >
          อีเมล
        </span>
        <input
          class="text-md w-full"
          required=""
          type="text"
          value=""
        />
      </div>
      <div
        class="my-2 w-full"
      >
        <span
          class="mb-2 text-sm"
        >
          รหัสผ่าน
        </span>
        <input
          class="text-md w-full"
          required=""
          type="password"
          value=""
        />
      </div>
      <div
        class="flex flex-col sm:flex-row mt-2 sm:mt-4"
      >
        <button
          class="w-full my-2 mr-0 sm:my-0 sm:mr-2"
        >
          เข้าสู่ระบบ
        </button>
        <button
          class="w-full my-2 ml-0 sm:my-0 sm:ml-2 bg-blue-400 hover:bg-blue-500"
        >
          สร้างบัญชีผู้ใช้
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
